python/mxnet/ndarray/random.py

```diff
@@ -518,3 +518,52 @@ def shuffle(data, **kwargs):
     <NDArray 2x3 @cpu(0)>
     """
     return _internal._shuffle(data, **kwargs)
+
+
+def randint(low, high, shape=_Null, dtype=_Null, ctx=None, out=None, **kwargs):
+    """Draw random samples from a discrete uniform distribution.
+
+    Samples are uniformly distributed over the half-open interval *[low, high)*
+    (includes *low*, but excludes *high*).
+
+    Parameters
+    ----------
+    low : int, required
+        Lower boundary of the output interval. All values generated will be
+        greater than or equal to low.
+    high : int, required
+        Upper boundary of the output interval. All values generated will be
+        less than high.
+    shape : int or tuple of ints, optional
+        The number of samples to draw. If shape is, e.g., `(m, n)` and `low` and
+        `high` are scalars, output shape will be `(m, n)`.
+    dtype : {'int32', 'int64'}, optional
+        Data type of output samples. Default is 'int32'
+    ctx : Context, optional
+        Device context of output. Default is current context. Overridden by
+        `low.context` when `low` is an NDArray.
+    out : NDArray, optional
+        Store output to an existing NDArray.
+
+
+    Examples
+    --------
+    >>> mx.nd.random.randint(5, 100)
+    [ 90]
+    <NDArray 1 @cpu(0)
+    >>> mx.nd.random.randint(-10, 2, ctx=mx.gpu(0))
+    [ -8]
+    <NDArray 1 @gpu(0)>
+    >>> mx.nd.random.randint(-10, 10, shape=(2,))
+    [ -5  4]
+    <NDArray 2 @cpu(0)>
+    >>> low = mx.nd.array([1,2,3])
+    >>> high = mx.nd.array([2,3,4])
+    >>> mx.nd.random.randint(low, high, shape=2)
```

Comment:
Is this example supported. can we add similar use cases to tests

Reply:
I was going to support arrays too. But had some issues. Hence removed it. Will update the example accordingly.